### ISSUE TITLE
How does iceberg ensure the correctness of data writing under high concurrency

### ISSUE THREAD

**Issue by huanghanyu-vungle** [apache/iceberg#6885]
### Query engine

query engine: spark 

### Question

 If multiple processes modify an iceberg table at the same time,  What method does iceberg use to ensure the correctness of data writing，Can 100% of the data written by different jobs not be lost?

For example:

I have two spark jobs，use spark.sql("merge into ...") in parallel, How to ensure that the data written to each other is 100% correct，what method does iceberg use to do this

thanks
